tests/Functional/TransactionTest.php

```diff
@@ -7,45 +7,55 @@
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Exception\ConnectionLost;
 use Doctrine\DBAL\Platforms\AbstractMySQLPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
 use Doctrine\DBAL\Schema\Table;
 use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
 use Doctrine\DBAL\Types\Types;
 
 use function func_get_args;
 use function restore_error_handler;
 use function set_error_handler;
-use function sleep;
+use function sprintf;
 
 use const E_WARNING;
 
 class TransactionTest extends FunctionalTestCase
 {
+    public function testBeginTransactionFailure(): void
+    {
+        $this->markConnectionNotReusable();
```

Comment:
Would it be fair to also move this call to `killCurrentSession()`? This way, there will be less code duplication and be even more obvious why it's not reusable.